### DISCUSSION TITLE
Small lag first time plugs are opened

### DISCUSSION THREAD

**Discussion by junocomp** [elementary/settings#329] [General]
There's a small lag when clicking any of the settings icons, such as a Desktop, About, Network, etc for the first time. The second/third time you click on the same settings the lag is gone and it works fast.

**Comment by cassidyjames** [comment]
This looks like it is because each plug is constructed the first time it is opened. I believe this is preferable to constructing every single plug when Switchboard is opened, and this is probably super dependent on hardware. For example, on my i7-5557U with 16 GB RAM it's barely noticeable, though with some plugs like Keyboard and Language & Region it's more noticeable.

**Comment by jeremypw** [comment]
Still noticeable in the Gtk4 port, resulting in visual glitch at start of animation.

**Comment by danirabbit** [comment]
I think the only practical solution here is to make sure each view constructs its contents in a more asynchronous way. So converting to a discussion since this isn't something that can be tracked as an individual issue. Issues need to be filed against each settings pain where we have slow loading
